src/rgw/rgw_data_sync.cc

```diff
@@ -1249,6 +1249,24 @@ struct rgw_sync_pipe_info_set {
   }
 };
 
+class LatencyConcurrencyControl : public LatencyMonitor {
+public:
+  CephContext* const cct;
+
+  LatencyConcurrencyControl(CephContext* cct)
+    : cct(cct)  {}
+  uint32_t get_concurrency() {
+    if (avg_latency() >= 10s) [[unlikely]] {
+      return cct->_conf->rgw_data_sync_spawn_window / 2;
+    } else if (avg_latency() >= 20s) [[unlikely]] {
+      return 1;
+    } else [[likely]] {
+      return cct->_conf->rgw_data_sync_spawn_window / 2;
```

Comment:
i see that you're using this in bucket sync as well, which has a separate `rgw_bucket_sync_spawn_window` configurable

is it really necessary to modify bucket sync? shouldn't data sync's throttling be enough here?

Reply:
I think we want to modify bucket sync, too, but I stuffed it into DataSyncCtx so we don't have to thread everything everywhere.